synapse/lib/aha.py

```diff
@@ -128,6 +131,31 @@ async def fini():
 
         self.onfini(fini)
 
+    async def initServiceRuntime(self):
+        self.addActiveCoro(self._clearInactiveSessions)
+
+    async def _clearInactiveSessions(self):
+        # Capture the current online svcs
+        online_svcs = [svc async for svc in self.getAhaSvcs() if svc.get('svcinfo', {}).get('online')]
+
+        # Capture current session idens
+        current_sessions = {s_common.guid(iden) for iden in self.dmon.sessions.keys()}
+
+        # Setup all our coroutines we need then await them to reduce race condition risks
+        futs = []
+        for svc in online_svcs:
+
+            svcname = svc.get('svcname')
+            network = svc.get('svcnetw')
+            linkiden = svc.get('svcinfo').get('online')
+            if linkiden not in current_sessions:
+                futs.append(self.setAhaSvcDown(svcname, linkiden, network=network))
+
+        await asyncio.gather(*futs)
```

Comment:
What benefit do we gain by making these gather?  Especially since the nexs stuff enforces that no two of them can be executing simultaneously?